hazelcast/src/main/java/com/hazelcast/util/CollectionUtil.java

```diff
@@ -92,4 +95,21 @@ public static boolean isNotEmpty(Collection collection) {
         }
         return item;
     }
+
+    /**
+     * @param collection           collection of objects to be converted to collection of data
+     * @param serializationService will be used for converting object to data
+     * @return collection of data
+     * @throws java.lang.NullPointerException if collection is  empty, null, or contains a null element
```

Comment:
seems there is no NPE when supplied collection is empty but javaDoc says opposite.